02.calendar/cal.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+require 'optparse'
+require 'date'
+
+TargetMonthMargin = 15 # 表示対象の月・年表示の余白
+DateMargin = 2 #カレンダーのDate間の余白
+
+options = {}
+
+# コマンドライン引数-m、-yで指定された値を月、年に代入する 
+opt = ARGV.getopts('m:','y:')
+options[:month] = opt['m'].to_i if opt['m']
+options[:year] = opt['y'].to_i if opt['y']
+
+# 引数が未指定だった場合、今月、今年を代入する
+options[:month] ||= Date.today.month
+options[:year] ||= Date.today.year
+
+# 指定された月、年のカレンダーを表示する
+puts "#{options[:month]}月 #{options[:year]}年".rjust(TargetMonthMargin)
+puts '日 月 火 水 木 金 土'
+
+# 指定された月の初日を取得する
+first_day = Date.new(options[:year], options[:month], 1)
```

Comment:
```suggestion
first_date = Date.new(options[:year], options[:month], 1)
```

Dateオブジェクトならdateと名付けた方がいいと思います
https://bootcamp.fjord.jp/pages/readable-variable-name

Reply:
`first_date` , `late_date` に修正しました。